types/pdfmake/interfaces.d.ts

```diff
@@ -0,0 +1,473 @@
+export type OrderedListElement = Content & {
+    counter?: number;
+    listType?: OrderedListType;
+};
+
+export type UnorderedListType = 'square' | 'circle' | 'none';
+export type UnorderedListElement = Content & {
+    listType?: UnorderedListType;
+};
+
+export type CanvasElement = CanvasRect | CanvasPolyline | CanvasLine | CanvasEllipse;
+
+export interface CanvasRect extends CanvasLineElement, CanvasFilledElement {
+    type: 'rect';
+    x: number;
+    y: number;
+    w: number;
+    h: number;
+    r?: number;
+}
+
+export interface CanvasPolyline extends CanvasLineElement, CanvasFilledElement {
+    type: 'polyline';
+    points: Array<{ x: number; y: number }>;
+    closePath?: boolean;
+    lineCap?: 'round' | 'square';
+}
+
+export interface CanvasLine extends CanvasLineElement {
+    type: 'line';
+    x1: number;
+    y1: number;
+    x2: number;
+    y2: number;
+    lineCap?: 'round' | 'square';
+}
+
+export interface CanvasEllipse extends CanvasLineElement, CanvasFilledElement {
+    type: 'ellipse';
+    x: number;
+    y: number;
+    r1: number;
+    r2?: number;
+}
+
+export interface CanvasFilledElement {
+    color?: string;
+    fillOpacity?: number;
+    linearGradient?: string[];
+}
+
+export interface CanvasLineElement {
+    lineWidth?: number;
+    lineColor?: string;
+    dash?: {
+        length: number;
+        space?: number;
+    };
+}
+
+export interface StyleDictionnary {
+    [name: string]: Style;
+}
+
+export type PDFVersion = '1.3' | '1.4' | '1.5' | '1.6' | '1.7' | '1.7ext3';
+
+export interface Watermark {
+    text: string;
+    opacity?: number;
+    angle?: number;
+    font?: string;
+    fontSize?: number;
+    color?: string;
+    bold?: boolean;
+    italics?: boolean;
+}
+
+export interface TDocumentDefinitions {
+    content: Content;
+    background?: DynamicBackground | Content;
+    compress?: boolean;
+    defaultStyle?: Style;
+    footer?: DynamicContent | Content;
+    header?: DynamicContent | Content;
+    images?: { [key: string]: string };
+    info?: TDocumentInformation;
+    pageBreakBefore?: (
+        currentNode: Node,
+        followingNodesOnPage: Node[],
+        nodesOnNextPage: Node[],
+        previousNodesOnPage: Node[],
+    ) => boolean;
+    pageMargins?: Margins;
+    pageOrientation?: PageOrientation;
+    pageSize?: PageSize;
+    styles?: StyleDictionnary;
+    userPassword?: string;
+    ownerPassword?: string;
+    permissions?: PDFKit.DocumentPermissions;
+    version?: PDFVersion;
+    watermark?: string | Watermark;
+}
+
+export interface Node {
+    text?: Content;
+    ul?: UnorderedListElement[];
+    ol?: OrderedListElement[];
+    table?: Table;
+    image?: string;
+    qr?: string;
+    canvas?: CanvasElement;
+    svg?: string;
+    columns?: Column[];
+    id?: string;
+    headlineLevel?: number;
+    style?: string | string[] | Style;
+    pageBreak?: 'before' | 'after';
```

Comment:
why not create a new type for this instead? That's how it was previously defined.

Reply:
You're right, I'll update it

Reply:
This looks good.